tech-papers/lbft-verification/lbft-verification.tex

```diff
@@ -0,0 +1,199 @@
+
+\subsection{Verifying the latest LedgerInfo with waypoints}
+\label{waypoint-li-verification}
+As described in Section~\ref{verification-epochs}, an \texttt{UpdateToLatestLedgerResponse} message carries a data structure for verifying validator changes, \texttt{ValidatorChangeProof}, which contains a list of all the epoch change \texttt{LedgerInfos}. A client verifies this list in an iterative fashion starting with its initial trusted \texttt{ValidatorSet}.
+
+\begin{figure}[ht]
+	\centering
+	\includegraphics[width=\textwidth]{figures/waypoint-verification.pdf}
+	\caption{\footnotesize{The first LedgerInfo in the \texttt{ValidatorChangeProof} is verified via a local waypoint.}}
+	\label{fig:waypoint-verification}
+\end{figure}
+
+Waypoint verification is working slightly differently as shown in Figure~\ref{fig:waypoint-verification}.
+A waypoint version corresponds to the last version of an epoch (version $0$ being somewhat special and corresponding to the last and only version of epoch $0$).
+Hence, the \texttt{ValidatorChangeProof} in the response starts with the end-of-epoch \texttt{LedgerInfo} corresponding to the waypoint: this very first \texttt{LedgerInfo} is verified using the waypoint. The rest of the epoch changes is verified in a regular manner.
+
+\subsection{State synchronization with waypoints}
+\label{waypoint-state-sync}
+Unfortunately, the verification of the latest \texttt{LedgerInfo} by itself might not be enough for secure state synchronization.
+Consider a following scenario:
+\begin{itemize}
+\item Full node $A$ has reached epoch $100$.
+\item Full node $B$ has $A$ configured as its upstream. $B$ latest known epoch is $10$ and it is restarted with waypoint $W_{80}$ corresponding to the last version of epoch $80$.
+\item Full node $C$ has $B$ configured as its upstream. $C$ latest known epoch is $10$ and it is restarted with waypoint $W_{60}$ corresponding to the last version of epoch $60$.
+\end{itemize}
+
+What is the right way for $B$ to synchronize its state to $A$? Given a waypoint $W_{80}$, $B$ can retrieve all the transaction chunks up to the end of epoch $80$ with proofs relative to the latest \texttt{LedgerInfo} of epoch $80$, however, that would not include the \texttt{LedgerInfos} for all the epoch changes prior to epoch $80$. As a result, $B$ would not be able to help node $C$ to synchronize its state from epoch $60$ to epoch $80$ because it doesn't have the \texttt{LedgerInfos} on epoch boundaries to prove the validator changes in range $60-80$.
+
+\begin{figure}[ht]
+	\centering
+	\includegraphics[width=\textwidth]{figures/state-sync-waypoint.pdf}
+	\caption{\footnotesize{ChunkResponse carries the LI of a waypoint and an optional intermediate end-of-epoch LI.}}
+	\label{fig:state-sync-waypoint}
+\end{figure}
+
+The solution is for $B$ to bring the intermediate end-of-epoch \texttt{LedgerInfos} in a verifiable manner using the following rules depicted in Figure~\ref{fig:state-sync-waypoint}:
```

Comment:
From my understanding, the difference between state sync across epoch and state sync with waypoints is not about bringing the intermediate end-of-epoch LedgerInfo(state sync across epoch brings them too) but the verification towards waypoints instead of the end-of-epoch LedgerInfo because of long range attack.

Imo this challenge is not described very clear  in this section, and the solution is not highlighted but just one of the bullet point below.
> The range of transactions in the response chunk is verified using the range merkle tree proof relativeto the version ofWaypointLedgerInfo.